sql/src/SqlEval.sk

```diff
@@ -1816,14 +1816,11 @@ private mutable class SQLContext(
     (newContext, underLockF) = try {
       f(this.context with {globals => this.globals})
     } catch {
-    | err @ SKJSON.ParseError _
-    | err @ SKJSON.FieldError _
-    | err @ SKJSON.SuperTypeError _
-    | err @ SKJSON.ReconstructError _
-    | err @ Error _ ->
-      (SKStore.cloneContextWithError(this.context, err), None())
-    | exn -> throw exn
+    | ex ->
+      cexn = SKStore.destroyObstackWithValue(pos, List[ex]).getHead();
+      throw cexn
```

Comment:
Do we want to always throw here? The code previously did not, and I suspect it might interfere with how errors are being bubbled up (although the previous clunky mechanism for bubbling errors up relied on setting a variable on the context since `destroyObstackWithValue()` was not available then).